crates/subspace-service/src/dsn/import_blocks/root_blocks.rs

```diff
@@ -80,36 +81,42 @@ impl RootBlockHandler {
             .get_closest_peers(PeerId::random().into())
             .await;
 
+        let mut peer_handles = Vec::new();
         // Acquire root blocks from peers.
         match get_peers_result {
             Ok(mut get_peers_stream) => {
                 while let Some(peer_id) = get_peers_stream.next().await {
-                    trace!(%peer_id, "get_closest_peers returned an item");
-
-                    let request_result = self
-                        .dsn_node
-                        .send_generic_request(
-                            peer_id,
-                            RootBlockRequest::LastRootBlocks {
-                                root_block_number: ROOT_BLOCK_NUMBER_PER_REQUEST,
-                            },
-                        )
-                        .await;
-
-                    match request_result {
-                        Ok(RootBlockResponse { root_blocks }) => {
-                            trace!(
-                                %peer_id,
-                                root_blocks_number=%root_blocks.len(),
-                                "Last root block request succeeded."
-                            );
-
-                            peer_blocks.insert(peer_id, root_blocks);
-                        }
-                        Err(error) => {
-                            debug!(%peer_id, ?error, "Last root block request failed.");
-                        }
-                    };
+                    let peer_blocks = peer_blocks.clone();
+                    let dsn_node = self.dsn_node.clone();
+                    let join_handle = tokio::spawn(async move {
```

Comment:
Hm... another algorithm change that should have been squashed...

On a different note this is problematic in library code because it is not going to be cancelled on drop, meaning user that drops the future to cancel everything, including dropping `node` will end up with `node` being alive, occupying ports, etc.

Create `FuturesUnordered` collection to do concurrency instead that can be dropped any time with all its contents, one CPU core should have more than sufficient CPU to drive all of these futures. As a bonus you'll not need `Arc<Mutex<>>` anymore.